omniledger/service/service.go

```diff
@@ -581,8 +588,35 @@ func (s *Service) verifySkipBlock(newID []byte, newSB *skipchain.SkipBlock) bool
 		log.Lvl2(s.ServerIdentity(), "State Changes hash doesn't verify")
 		return false
 	}
-	// TODO compute new state and check if there's a state change on the
-	// config, if there is, then we check whether newSB matches the config
+
+	// Compute the new state and check whether the roster in newSB matches
+	// the config.
+	collClone := s.getCollection(newSB.SkipChainID()).coll.Clone()
+	for _, sc := range scs {
+		if err := storeInColl(collClone, &sc); err != nil {
+			log.Error(err)
+			return false
+		}
+	}
+	config, err := LoadConfigFromColl(&roCollection{collClone})
+	if err != nil {
+		log.Error(err)
+		return false
+	}
+	if !config.Roster.ID.Equal(newSB.Roster.ID) {
+		log.Error("rosters have unequal IDs")
+		return false
+	}
+	if len(config.Roster.List) != len(newSB.Roster.List) {
```

Comment:
Is this `if` and the following `for` not covered in `Roster.ID.Equal`?

Reply:
From looking at the code, `Roster.ID.Equal` only checks the uuid. It possible to use the same ID but a different list of server identities to "fool" the verification function if we don't check individual server identities. In this case, the ID will be wrong and an error might show up later, but I just thought it's a good idea to catch problems early.